### PR TITLE
Throw an error if Target.Dependency.productItem is used instead of .product. 

### PR DESCRIPTION
**Pull request by elsh**
Target.Dependency.productItem/targetItem/byNameItem should not be used in a package manifest (they are only visible/public since they are part of function sigs that became public). Changing them to internal requires a lot more changes so throw an error with a suggestion to use the right API for now. Creating a static func for targetItem and byNameItem throws an invalid redeclaration error so this PR currently only handles .productItem use case.
Resolves rdar://91013735

### CODE REVIEW

**Comment by elsh** [comment]
@swift-ci smoke test

File: Sources/PackageDescription/Target.swift [hunk 1]
```diff
@@ -922,6 +922,11 @@ public final class Target {
 }
 
 extension Target.Dependency {
+    @available(_PackageDescription, obsoleted: 5.6, message: "use .product(name:package:condition) instead.")
```
Comment (neonichu) [inline Sources/PackageDescription/Target.swift:925]: I think this should be 5.7, we already shipped 5.6

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by neonichu** [comment]
Thanks, LGTM. 

Since this API was never supposed to be public in the first place and developers can upgrade to a newer tools-version on their own pace, it seems right to directly obsolete this without a deprecation period.

**Review by neonichu — APPROVED** [review]

**Review by abertelrud — APPROVED** [review]

**Pull request merged**